diff --git a/index.js b/index.js
index 5d972c6ce3d83dc808e8f74b3c01c13a64fcf002..fc63d264e25c49610c83d9b25b84164130621f95 100644
--- a/index.js
+++ b/index.js
@@ -15,6 +15,7 @@ import {
   symlinkSync,
   writeFileSync
 } from "fs";
+import { normalizePath } from "vite";
 import mm from "micromatch";
 import { dirname, join, relative } from "path";
 import process from "process";
@@ -132,9 +133,9 @@ export default function ({ edge, prerender, includes, excludes } = {}) {
       // Vercel Build Output API v3 (https://vercel.com/docs/build-output-api/v3)
       const __dirname = dirname(fileURLToPath(import.meta.url));
       const workingDir =
-        config.root === process.cwd()
+        config.root === normalizePath(process.cwd())
           ? pathToFileURL(config.root + "/")
-          : new URL(config.root, pathToFileURL(process.cwd() + "/"));
+          : new URL(config.root, pathToFileURL(normalizePath(process.cwd()) + "/"));console.log(workingDir, config.root === normalizePath(process.cwd()), config.root, normalizePath(process.cwd()));
       const vercelOutputDir = new URL("./.vercel/output/", workingDir); // join(config.root, ".vercel/output");
       const outputDir = new URL("./dist/", workingDir); // join(config.root, ".vercel/output");
       const solidServerDir = new URL("./.solid/server/", workingDir); //  join(config.root, "./.solid/server/");